src/systems/implicit_system.C

```diff
@@ -711,7 +711,7 @@ void ImplicitSystem::adjoint_qoi_parameter_sensitivity (const QoISet & qoi_indic
   // if we havent already or dont have an initial condition for the adjoint
   if (!this->is_adjoint_already_solved())
     {
-      this->adjoint_solve(qoi_indices);
+      this->ImplicitSystem::adjoint_solve(qoi_indices);
```

Comment:
Why don't we want to let derived systems override this?

Reply:
The this pointer here can be for a TS object.

Reply:
The this pointer in ImplicitSystem::anything() is always going to be an ImplicitSystem or a subclass of one, which TimeSolver isn't.

Reply:
Yeah, I guess went overboard with the disambiguation here.